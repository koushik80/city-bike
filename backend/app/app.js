// external imports
const express = require("express");
const app = express();
const dotenv = require("dotenv");
const cors = require("cors");

// dot environment config
if (process.env.NODE_ENV !== "PRODUCTION") {
    dotenv.config({ path: 'backend/config/config.env' });
}

// parsing json
app.use(express.json());

// used cors to allow cross-origin access with frontend
app.use(cors());

// imported routers (internal imports)
const journey = require("../routes/journey.routes");
const station = require("../routes/station.routes");
const { errorMiddleware } = require("../middleware/errorMiddleware");

// middleware
app.use("/api/journey", journey);
app.use("/api/station", station);


app.get('/', (req, res) => {
    res.status(200).json({
        message: "This is JOURNEY REST API"
    });
});

// ERROR MIDDLEWARE
app.use(errorMiddleware)

module.exports = app;