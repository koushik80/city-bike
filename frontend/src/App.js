import React, { Suspense } from 'react';
import Routes from './Routes';

const App = () => {
  return (
    <Suspense fallback={<h1>Site Loading...</h1>}>
      <Routes />
    </Suspense>
  );
};

export default App;
