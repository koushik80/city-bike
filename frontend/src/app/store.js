// https://redux-toolkit.js.org/rtk-query/overview

import { configureStore } from '@reduxjs/toolkit'
import apiSlice from '../features/api/apiSlice'
import filterReducer from '../features/filter/filterSlice'

export const store = configureStore({
  reducer: {
    [apiSlice.reducerPath]: apiSlice.reducer,
    filter: filterReducer
  },
  devTools: process.env.NODE_ENV !== 'production',
  middleware: (getDefaultMiddlewares) => getDefaultMiddlewares().concat(apiSlice.middleware)
})