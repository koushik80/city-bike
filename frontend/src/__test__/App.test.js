import React from 'react'
import { Provider } from 'react-redux'
import { store } from '../app/store'
import App from './../App'
import renderer from 'react-test-renderer'

describe('renders app component', () => {

  it('App component match snapshot', () => {

    const wrapper = renderer.create(<Provider store={store}><App /></Provider>).toJSON()
    expect(wrapper).toMatchSnapshot()
  })
})