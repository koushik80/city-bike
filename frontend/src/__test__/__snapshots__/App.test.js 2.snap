// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders app component App component match snapshot 1`] = `
Array [
  <h1>
    Site Loading...
  </h1>,
]
`;
